airbyte-integrations/connectors/source-yandex-metrica/source_yandex_metrica/spec.yaml

```diff
@@ -0,0 +1,292 @@
+          - ym:pv:browserMajorVersion
+          - ym:pv:browserMinorVersion
+          - ym:pv:browserCountry
+          - ym:pv:browserEngine
+          - ym:pv:browserEngineVersion1
+          - ym:pv:browserEngineVersion2
+          - ym:pv:browserEngineVersion3
+          - ym:pv:browserEngineVersion4
+          - ym:pv:browserLanguage
+          - ym:pv:clientTimeZone
+          - ym:pv:cookieEnabled
+          - ym:pv:deviceCategory
+          - ym:pv:from
+          - ym:pv:hasGCLID
+          - ym:pv:GCLID
+          - ym:pv:ipAddress
+          - ym:pv:javascriptEnabled
+          - ym:pv:mobilePhone
+          - ym:pv:mobilePhoneModel
+          - ym:pv:openstatAd
+          - ym:pv:openstatCampaign
+          - ym:pv:openstatService
+          - ym:pv:openstatSource
+          - ym:pv:operatingSystem
+          - ym:pv:operatingSystemRoot
+          - ym:pv:physicalScreenHeight
+          - ym:pv:physicalScreenWidth
+          - ym:pv:regionCity
+          - ym:pv:regionCountry
+          - ym:pv:regionCityID
+          - ym:pv:regionCountryID
+          - ym:pv:screenColors
+          - ym:pv:screenFormat
+          - ym:pv:screenHeight
+          - ym:pv:screenOrientation
+          - ym:pv:screenWidth
+          - ym:pv:windowClientHeight
+          - ym:pv:windowClientWidth
+          - ym:pv:lastTrafficSource
+          - ym:pv:lastSearchEngine
+          - ym:pv:lastSearchEngineRoot
+          - ym:pv:lastAdvEngine
+          - ym:pv:artificial
+          - ym:pv:pageCharset
+          - ym:pv:isPageView
+          - ym:pv:link
+          - ym:pv:download
+          - ym:pv:notBounce
+          - ym:pv:lastSocialNetwork
+          - ym:pv:httpError
+          - ym:pv:clientID
+          - ym:pv:networkType
+          - ym:pv:lastSocialNetworkProfile
+          - ym:pv:goalsID
+          - ym:pv:shareService
+          - ym:pv:shareURL
+          - ym:pv:shareTitle
+          - ym:pv:iFrame
+          - ym:pv:parsedParamsKey1
+          - ym:pv:parsedParamsKey2
+          - ym:pv:parsedParamsKey3
+          - ym:pv:parsedParamsKey4
+          - ym:pv:parsedParamsKey5
+          - ym:pv:parsedParamsKey6
+          - ym:pv:parsedParamsKey7
+          - ym:pv:parsedParamsKey8
+          - ym:pv:parsedParamsKey9
+          - ym:pv:parsedParamsKey10
+      default:
+        - ym:pv:watchID        
+        - ym:pv:dateTime
+      required:
+        - ym:pv:watchID
+        - ym:pv:dateTime
+      order: 4
+    visits_fields:
+      title: Visits Fields
+      description: Fields for sessions data. See https://yandex.com/dev/metrika/doc/api2/logs/fields/visits.html
+      type: array
+      items:
+        type: string
+        enum:
+          - ym:s:visitID
+          - ym:s:counterID
+          - ym:s:watchIDs
+          - ym:s:date
+          - ym:s:dateTime
+          - ym:s:dateTimeUTC
+          - ym:s:isNewUser
+          - ym:s:startURL
+          - ym:s:endURL
+          - ym:s:pageViews
+          - ym:s:visitDuration
```

Comment:
Same here. Adding this complexity to the user. Maybe change the discover step to retrieve dynamically the schema as other connectors.

Reply:
Removed field choices during setup. Fields are passed from a dictionary in fields.py